### ISSUE TITLE
Logits and Weights Wi

### ISSUE THREAD

**Issue by Styleoshin** [tensorflow/tensor2tensor#1612]
Hey, i readed the code source, and i founded somes difficulties to find an anwsers.
1) The weights for the Wik, Wik and Wik matrices. The matrices will be formed before the training of the model or the values ​​will be generated randomly and their formation will be in same time with the training of the model.
2) How this matrices are trained ?
3) During the training of the model (transformer), which loss function is used to determine the probability distributions in the Logits vector ? (cross entropy or kl_divergence)
Thanks

**Comment by lukaszkaiser** [comment]
These are very generic questions, let's stick to code issues here.
